Doc updates about runtime classpath and main class (#3292)
GoogleContainerTools/jib: jib-maven-plugin/README.md

`entrypoint` | list | *None* | The command to start the container with (similar to Docker's [ENTRYPOINT](https://docs.docker.com/engine/reference/builder/#entrypoint) instruction). If set, then `jvmFlags`, `mainClass`, `extraClasspath`, and `expandClasspathDependencies` are ignored. You may also set `<entrypoint>INHERIT</entrypoint>` (`<entrypoint><entry>INHERIT</entry></entrypoint>` in old Maven versions) to indicate that the `entrypoint` and `args` should be inherited from the base image.\*
`environment` | map | *None* | Key-value pairs for setting environment variables on the container (similar to Docker's [ENV](https://docs.docker.com/engine/reference/builder/#env) instruction).
`extraClasspath` | list | *None* | Additional paths in the container to prepend to the computed Java classpath.

> `expandClasspathDependencies` | boolean | `false` | <ul><li>Java 8 *or* Jib < 3.1: When set to true, does not use a wildcard (for example, `/app/lib/*`) for dependency JARs in the default Java runtime classpath but instead enumerates the JARs. Has the effect of preserving the classpath loading order as defined by the Maven project.</li><li>Java >= 9 *and* Jib >= 3.1: Jib always enumerates the dependencies JARs. (Therefore, the option has no effect.) This is achieved by [creating and using a JVM argument file](#custom-container-entrypoint) for the `--class-path` JVM argument.</li></ul>

Reviewer: What do you think of this?
```suggestion
`expandClasspathDependencies` | boolean | `false` | <ul><li>Java 8 and >=9 before Jib 3.1: When set to true, does not use a wildcard (for example, `/app/lib/*`) for dependency JARs in the default Java runtime classpath but instead enumerates the JARs. Has the effect of preserving the classpath loading order as defined by the Maven project.</li><li>Java >= 9 *and* Jib >= 3.1: This option has no effect. Jib *always* enumerates the dependencies JARs.  This is achieved by [creating and using a JVM argument file](#custom-container-entrypoint) for the `--class-path` JVM argument.</li></ul>
```

Author: I like the suggestion for the second bullet point.

For the first one, I think `Java 8 and >=9 before Jib 3.1` is ambiguous, in that it could be read like `(Java 8 and >= 9) before Jib 3.1`. In any case, I'll merge this to now and apply the suggestion in a later PR. Many doc changes (including Stack Overflow) depend on this. I'll make sure to update this.

Author: Will be done by #3293.
